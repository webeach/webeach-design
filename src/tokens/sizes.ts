export const Size = {
  '1x': '4px',
  '1.5x': '6px',
  '2x': '8px',
  '3x': '12px',
  '4x': '16px',
  '4.5x': '18px',
  '5x': '20px',
  '6x': '24px',
  '7x': '28px',
  '8x': '32px',
  '10x': '40px',
  '12x': '48px',
  '16x': '64px',
  '24x': '96px',
  '32x': '128px',
} as const;

export const BorderSize = {
  thin: '1px',
  regular: '2px',
  medium: Size['1x'],
  bold: Size['1.5x'],
  extraBold: Size['2x'],
} as const;

export const Rounding = {
  '1x': Size['1x'],
  '1.5x': Size['1.5x'],
  '2x': Size['2x'],
  '4x': Size['4x'],
} as const;

export const Spacing = {
  '0.5x': '2px',
  '1x': Size['1x'],
  '1.5x': Size['1.5x'],
  '2x': Size['2x'],
  '3x': Size['3x'],
  '4x': Size['4x'],
  '5x': Size['5x'],
  '6x': Size['6x'],
  '7x': Size['7x'],
  '8x': Size['8x'],
  '10x': Size['10x'],
  '12x': Size['12x'],
} as const;

export type SizeType = Readonly<typeof Size>;
export type SizeKey = keyof SizeType;
export type SizeValue = SizeType[SizeKey];

export type BorderSizeType = Readonly<typeof BorderSize>;
export type BorderSizeKey = keyof BorderSizeType;
export type BorderSizeValue = BorderSizeType[BorderSizeKey];

export type RoundingType = Readonly<typeof Rounding>;
export type RoundingKey = keyof RoundingType;
export type RoundingValue = RoundingType[RoundingKey];

export type SpacingType = Readonly<typeof Spacing>;
export type SpacingKey = keyof SpacingType;
export type SpacingValue = SpacingType[SpacingKey];
