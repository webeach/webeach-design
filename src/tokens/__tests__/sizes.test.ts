import { BorderSize, Rounding, Size, Spacing } from '../sizes';

import { borderSizeMapSchema } from './schemas/borderSizeMapSchema';
import { sizeMapSchema } from './schemas/sizeMapSchema';

describe('Sizes validation', () => {
  test('Validate Size', () => {
    expect(() => sizeMapSchema.parse(Size)).not.toThrow();
  });

  test('Validate BorderSize', () => {
    expect(() => borderSizeMapSchema.parse(BorderSize)).not.toThrow();
  });

  test('Validate Rounding', () => {
    expect(() => sizeMapSchema.parse(Rounding)).not.toThrow();
  });

  test('Validate Spacing', () => {
    expect(() => sizeMapSchema.parse(Spacing)).not.toThrow();
  });

  test('Snapshot Size', () => {
    expect(Size).toMatchSnapshot();
  });

  test('Snapshot BorderSize', () => {
    expect(BorderSize).toMatchSnapshot();
  });

  test('Snapshot Rounding', () => {
    expect(Rounding).toMatchSnapshot();
  });

  test('Snapshot Spacing', () => {
    expect(Spacing).toMatchSnapshot();
  });
});
