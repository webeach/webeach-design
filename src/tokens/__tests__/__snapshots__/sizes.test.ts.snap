// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sizes validation Snapshot BorderSize 1`] = `
{
  "bold": "6px",
  "extraBold": "8px",
  "medium": "4px",
  "regular": "2px",
  "thin": "1px",
}
`;

exports[`Sizes validation Snapshot Rounding 1`] = `
{
  "1.5x": "6px",
  "1x": "4px",
  "2x": "8px",
  "4x": "16px",
}
`;

exports[`Sizes validation Snapshot Size 1`] = `
{
  "1.5x": "6px",
  "10x": "40px",
  "12x": "48px",
  "16x": "64px",
  "1x": "4px",
  "24x": "96px",
  "2x": "8px",
  "32x": "128px",
  "3x": "12px",
  "4.5x": "18px",
  "4x": "16px",
  "5x": "20px",
  "6x": "24px",
  "7x": "28px",
  "8x": "32px",
}
`;

exports[`Sizes validation Snapshot Spacing 1`] = `
{
  "0.5x": "2px",
  "1.5x": "6px",
  "10x": "40px",
  "12x": "48px",
  "1x": "4px",
  "2x": "8px",
  "3x": "12px",
  "4x": "16px",
  "5x": "20px",
  "6x": "24px",
  "7x": "28px",
  "8x": "32px",
}
`;
